### PR TITLE
feat(shifts): department coverage pies on /Shifts

### PR DESCRIPTION
**Pull request by FrankFanteev**
## Summary

Adds a row of per-department coverage pies above the /Shifts browse page. Each pie shows the team's filled / requested hour ratio as a conic-gradient disc; clicking a pie filters the page to that department, clicking again clears the filter. A "Filter: X ×" chip appears between the filter form and the pie row when a department is selected.

The service (IShiftManagementService.GetDepartmentCoveragePiesAsync) lives in Humans.Application/Services/Shifts/, goes through IShiftManagementRepository + ITeamService.GetByIdsWithParentsAsync, and never imports EF Core. Pie eligibility is Team.IsInDirectory (top-level OR promoted sub-team); non-promoted sub-team rotas roll up to the parent. AdminOnly shifts and hidden rotas are excluded; all-day shifts use the 08:00–18:00 window, not the Duration column.

## What's new on the page

- Tip pill above the row: "click a chart to filter to that department; click it again to clear." Dismissable with persistence in localStorage (key humans.pieTipDismissed).
- Severity color on the disc: green ≥50%, amber 25–50%, rust <25%.
- 100% fade: completed teams fade to opacity .55 so the row visually de-emphasizes "done" departments. Hover and selected state restore full opacity.
- Sub-team grouping: promoted sub-teams render immediately after their parent in the row (display ordering applied in the page builder per memory/architecture/display-sort-in-controllers). Sub-teams carry a 3px gold border-inline-start and italic name.
- Filter chip under the filter form when a department is selected.
- A11y: row is role="group" with aria-label; pies are submit buttons with descriptive aria-label and title.
- i18n: en, es, ca, it, fr, de — all new strings localized.

## Architecture / rule compliance

- [SurfaceBudget(48 → 49)] on IShiftManagementService. History trimmed to 3 newest bullets per memory/code/surface-budget-history-trim.
- Display ordering is applied in the page builder (ShiftBrowsePageBuilder.OrderPiesGroupedByParent), not the service.
- Icon uses FontAwesome 6 (fa-solid fa-circle-info), not Bootstrap Icons (memory/code/icons-fa6-only).
- New CSS hooks into the existing design-token system (--h-*) — no --bs-* usage.

## Docs

- docs/features/shifts/department-coverage-pies.md — feature spec with user stories, eligibility, hour math, ordering.
- docs/sections/Shifts.md — section invariants block for the pie row (eligibility, exclusion rules, cap, all-day window, ordering location).

## Tests

- 13 service-level pie tests in tests/Humans.Application.Tests/Services/Shifts/ShiftManagementServiceCoveragePiesTests.cs covering: empty event, single-dept, promoted vs non-promoted sub-team rollup, hidden rota / AdminOnly exclusion, cap, all-day window, date-range filter, FillPercent rounding/zero-divisor, parent-only-via-subteam edge.
- 3 page-builder tests in tests/Humans.Web.Tests/Shifts/ShiftBrowsePageBuilderPieSortTests.cs for the grouped sort.

Full Application (2964) and Web (104) tests pass.

## Test plan

- [x] Open /Shifts as a volunteer with active event and >0 rotas — pies row renders above the period buttons.
- [x] Click a pie — page filters; selected pie has gold outline.
- [x] Click the same pie again — filter clears.
- [x] Click a different pie — filter switches.
- [x] Click the chip ✕ — clears just the department filter; period / dates / tags / sort preserved.
- [x] Click ✕ on the tip pill — pill disappears; survives reload.
- [x] Visit /Shifts for an event with no rotas — no pie row renders (early-return).
- [x] As an admin / coordinator, confirm AdminOnly shifts and hidden rotas don't contribute to pie hours.
- [x] In all 6 locales, the pie tooltip, ARIA label, tip text, dismiss label, and chip prefix render in the active language.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by peterdrier** [comment]
## PR Review — 2026-05-16

**PR**: #602 — feat(shifts): department coverage pies on /Shifts (https://github.com/peterdrier/Humans/pull/602)
**Branch**: pie-info-pill → main (head 53dc91a77)
**Scope**: 18 files changed, 6 commits, +1,181 / -9 LOC. Author: @FrankFanteev.

---

### Prior Review Status

Two prior bot (`github-actions`) inline findings on `0bddb0623`. Re-checked against head:

- `github-actions[bot]` flagged: `IShiftManagementService.cs:20` surface-budget history bullet has unfilled `peterdrier#?` PR placeholder — **STILL OPEN** at head (line 20 still reads `peterdrier#?`).
- `github-actions[bot]` flagged: `ShiftManagementService.cs:1033` `TeamService.GetByIdsWithParentsAsync` call drops `ct` — **STILL OPEN** at head; line 1031 still calls `GetByIdsWithParentsAsync(teamIdsWithRotas)` with no token, even though the very next line (1033) propagates `ct` on the rota fetch.

---

### Spec Compliance

No linked GitHub issue (PR body has no `Closes/Fixes/Resolves #N`). Self-assessed against the PR body's own acceptance criteria:

- **FULLY ADDRESSED**: pie row above /Shifts, click-to-filter / click-again-to-clear, gold-outline selection, dismissable tip pill with localStorage persistence, severity coloring (green/amber/rust), 100% fade with hover/selected restore, sub-team border + italic, filter chip, ARIA + `role="group"`, 6-locale strings, FA6 icon, `--h-*` design tokens (no `--bs-*`), display ordering in builder (not service), `[SurfaceBudget(48→49)]` with 3-entry trimmed history.
- **PARTIALLY ADDRESSED**: nothing in the body went uncovered.

`docs/features/shifts/department-coverage-pies.md` and the `docs/sections/Shifts.md` invariant block are present and aligned with the implementation.

---

### Code Quality

**IMPORTANT**

1. **`src/Humans.Application/Services/Shifts/ShiftManagementService.cs:1031` — CancellationToken dropped on `TeamService.GetByIdsWithParentsAsync` call.** Every other async call in `GetDepartmentCoveragePiesAsync` (`GetEventSettingsByIdAsync`, `GetTeamIdsWithRotasInEventAsync`, `GetRotasWithShiftsAndSignupsAsync`) passes `ct`; this one silently drops it. `ITeamService.GetByIdsWithParentsAsync` accepts a `CancellationToken` (Interfaces/Teams/ITeamService.cs:602). If the HTTP request is cancelled mid-method the team-lookup hop keeps running. Fix: `await TeamService.GetByIdsWithParentsAsync(teamIdsWithRotas, ct);`. *(Reiterating the bot finding — still open at head.)*

2. **`src/Humans.Application/Interfaces/Shifts/IShiftManagementService.cs:20` — surface-budget history bullet has unfilled `peterdrier#?` placeholder.** Per `memory/code/surface-budget-history-trim.md`, the PR reference is the audit-trail linking the budget bump to the authorising PR; every prior entry uses a real number (`nobodies-collective/Humans#700`). Replace `peterdrier#?` with `peterdrier#602`. *(Reiterating the bot finding — still open at head.)*

**MINOR**

3. **`src/Humans.Web/Views/Shifts/_DepartmentCoveragePieRow.cshtml:33-34` — `RequestedHours` / `FilledHours` formatted with `"0.#"` ignores locale.** The page is localized into 6 locales, including es/fr/de/it/ca where decimal separators are commas. `decimal.ToString("0.#")` always uses the invariant period. The cap is whole-hour-multiples in practice (Duration × Max), so values are almost always integers and the issue rarely materialises, but the literal `" h / "` separator and number formatting won't match the user's locale. If you want locale-correct formatting, pass `CultureInfo.CurrentUICulture` or use a localized format string. Acceptable to defer.

4. **`src/Humans.Web/Views/Shifts/_DepartmentCoveragePieRow.cshtml:53` — `--pct: @pct%;` interpolation.** Razor will HTML-encode the value, so `style="--pct: 33%;"` renders cleanly with safe integer input. No XSS concern (`pct` is `int`), but worth noting for future maintainers that the safety depends on the input being a primitive.

5. **`src/Humans.Web/Views/Shifts/_DepartmentCoveragePieRow.cshtml:63-76` — inline `<script>` not nonced.** The rest of the page uses inline `<script>` blocks the same way, so this is consistent with the existing pattern, not a regression. If a CSP nonce policy is introduced later, this and the existing scripts above will all need the same treatment.

6. **`src/Humans.Web/Models/Shifts/ShiftBrowsePageBuilder.cs:107-109` — `GetDepartmentCoveragePiesAsync` call doesn't pass a `CancellationToken`.** The builder method (`BuildAsync`) doesn't take one either, so this is consistent with the surrounding code, not a new gap. Flagging for awareness — if the call-chain is ever plumbed for cancellation, this site is one of the holes.

7. **Test coverage** is strong: 13 service-level pie tests cover empty event, single/promoted/non-promoted sub-team rollup, parent-only-via-subteam, AdminOnly + hidden exclusion, MaxVolunteers cap, all-day window, date filter, alphabetical sort, FillPercent rounding/zero-divisor. 3 builder tests cover the group-by-parent display sort. No gaps worth blocking on.

---

### Bottom Line

The feature is well-scoped, correctly layered (no EF in Application service, display sort in builder, FA6 icon, `--h-*` tokens, [SurfaceBudget] bumped with trimmed history), thoroughly tested (16 new tests covering the math edge cases) and fully localized into 6 locales. Two prior bot findings (CancellationToken drop on the `TeamService` call, unfilled `peterdrier#?` history placeholder) are still open at head and should be fixed before merge — both are one-line edits. The minor formatting / locale notes can be deferred.

---
*Review by Claude Code `/pr-review`*

**Comment by peterdrier** [comment]
## Review items addressed — commit 51791df53

| # | Item | Status |
|---|------|--------|
| 1 | `ShiftManagementService.cs:1031` — `TeamService.GetByIdsWithParentsAsync` missing `ct` | **FIXED** — plumbed `ct` through |
| 2 | `IShiftManagementService.cs:20` — `peterdrier#?` placeholder in `[SurfaceBudget]` history | **FIXED** — replaced with `peterdrier#602` |
| 3 | `_DepartmentCoveragePieRow.cshtml:32` — `--pct` CSS interpolation guard | **FIXED** — `Math.Clamp(pie.FillPercent, 0, 100)` applied before interpolation |
| 4 | Inline `<script>` consistency | **NO CHANGE** — already matches surrounding patterns in `Index.cshtml` |
| 5 | `ShiftBrowsePageBuilder.BuildAsync` — CancellationToken not plumbed | **FIXED** — added `ct = default` param, passed to `GetDepartmentCoveragePiesAsync`, controller passes `HttpContext.RequestAborted` |
| – | Locale-blind decimal formatting (`"0.#"`) | **SKIPPED** — per Peter, deferred |
| – | Test coverage observation | **SKIPPED** — informational only |

Build and all non-integration tests pass clean. Integration test failures are pre-existing (62 failures on base commit).

**Comment by peterdrier** [comment]
## Push correction

The previous fix comment said `git push origin HEAD:pie-info-pill`, but this PR's head is on `FrankFanteev/Humans`, not `peterdrier/Humans` — so the commit landed on the wrong remote and didn't show on the PR.

Re-pushed `51791df53` to `FrankFanteev/Humans:pie-info-pill`. PR head is now `51791df53` and the fixes are visible here.

---
*Correction by Claude Code `/pr-review`*

File: src/Humans.Application/Services/Shifts/ShiftManagementService.cs [hunk 1]
```diff
@@ -1009,6 +1014,81 @@ public Task<IReadOnlyList<Guid>> GetTeamIdsWithShiftsInEventAsync(
...
+        if (es is null) return [];
+
+        var teamIdsWithRotas = await _repo.GetTeamIdsWithRotasInEventAsync(eventSettingsId, ct);
+        if (teamIdsWithRotas.Count == 0) return [];
+
+        // Brings rota-owning teams AND their parents into one lookup, so a
+        // non-promoted sub-team rota can find its parent without a second hop.
+        var teamLookup = await TeamService.GetByIdsWithParentsAsync(teamIdsWithRotas);
+
+        var allRotas = await _repo.GetRotasWithShiftsAndSignupsAsync(
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/Shifts/ShiftManagementService.cs:1033]: Fixed in 51791df53 — `ct` now passed through to `TeamService.GetByIdsWithParentsAsync`.

File: src/Humans.Application/Interfaces/Shifts/IShiftManagementService.cs [hunk 2]
```diff
@@ -17,16 +17,12 @@ public record ShiftTagPreferenceSummary(Guid Id, string Name);
 /// <remarks>
 /// Surface-budget recent history (newest first):
 /// <list type="bullet">
+///   <item>48→49 — feat(shifts): department coverage pies on /Shifts (peterdrier#?). Added GetDepartmentCoveragePiesAsync. Authorized by Peter, 2026-05-16.</item>
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Interfaces/Shifts/IShiftManagementService.cs]: Fixed in 51791df53 — placeholder filled in as `peterdrier#602`.
